File: csharp/Original/PublishingHouse.Application/Books/BooksQueryService.cs
using PublishingHouse.Books.DTOs;
using PublishingHouse.Books.Entities;
using PublishingHouse.Books.Repositories;

namespace PublishingHouse.Application.Books;

public class BooksQueryService: IBookQueryService
{
    public Task<BookDetails?> FindDetailsById(BookId bookId, CancellationToken ct) =>
        repository.FindDetailsById(bookId, ct);

    public BooksQueryService(IBooksQueryRepository repository) =>
        this.repository = repository;

    private readonly IBooksQueryRepository repository;
}

File: node.js/src/original/application/books/booksQueryService.ts
import { BookDetails } from 'src/original/domain/books/dtos';
import { FindDetailsByIdQuery } from './queries';
import { IBooksQueryRepository } from 'src/original/domain/books/repositories';

export interface IBooksQueryService {
  findDetailsById(query: FindDetailsByIdQuery): Promise<BookDetails | null>;
}

export class BooksQueryService implements IBooksQueryService {
  constructor(private repository: IBooksQueryRepository) {}

  findDetailsById(query: FindDetailsByIdQuery): Promise<BookDetails | null> {
    return this.repository.findDetailsById(query.data.bookId);
  }
}
